lib/AL_USDMaya/AL/usdmaya/Utils.cpp

```diff
@@ -59,7 +59,6 @@ void matrixToSRT(GfMatrix4d& value, double S[3], MEulerRotation& R, double T[3])
   if(isNegated)
   {
     zAxis = -zAxis;
-    scaleZ = -scaleZ;
```

Comment:
The previous code aimed to handle negative scaling. Starting with an identity matrix and applying a scaling of [1, 1, -1], we get:
```
 1  0  0  0
 0  1  0  0
 0  0 -1  0
 0  0  0  1
```
By taking the length of the x/y/z axes, we end up with a scaling of [1, 1, 1], which is clearly wrong. The cross product tests to see if the expected orthogonal vectors match the vector in the z axis, i.e. 
```
if(x cross y != z)
```
There are 2 reasons for this.
1. Pretty much every matrix->quat, or matrix->euler will fail if the matrix is the wrong handedness. 
2. To figure out if we have a negative scaling. 

So if the handedness of the matrix is wrong, we need to flip an axis. Sadly without some prior information about the original SRT transforms, this incorrect handedness could be the result of:

1. A negative scaling in x
2. A negative scaling in y
3. A negative scaling in z
4. A negative scaling in x, y, _and_ z. 

Sadly there is no way to know which of those 4 scenarios is the one that created the scaling matrix. (A negative scaling in 2 axes is effectively just a rotation). Since there is no way to know _which_ of the 4 cases is correct, we end up needing to pick one, and this code basically chooses the 3rd option. So if we negate the z axis, that ensures the matrix->quat conversion works; however we also need to negate the z scaling, so that when we convert from SRT back to a matrix, the flipped handedness is re-introduced.

Reply:
This makes sense - however, this change was flagged by the clang static checker, because the eliminated line does nothing - the scaleZ is never used after this point.  Perhaps this block needs to be moved up?

Reply:
Yeah - so, on closer inspection, you're totally right, both are needed... but as it was written, the scaleZ inversion wasn't having an effect, so we need to move it up.  I'll do that in the morning, as well as write a test to make sure that we get the desired result.

Reply:
So, I confirmed that this command currently doesn't handle one-axis-negative-scaling correctly; however, looking at it a bit closer, there's another problem, which is that it doesn't correctly decompose the a full maya xform stack - in particular, it doesn't handle shear.

While you can make the case that shear isn't very frequently used by itself; however, it DOES pop up more frequently when you collapse a multiple-xform stack, which features non-uniform scaling.  (ie, if you make an xform with several parents above it, and all the various xforms feature translation, rotation, and non-uniform scaling, and then unparent the child to the world, preserving the xform, you will get shear).

Thankfully, I think there's an easy fix here - simply use maya's built-in matrix decomposition operations.  I think I'll just leave this PR as is (since, as it is, it at least doesn't change the behavior), and make a new PR to fix this.